### DISCUSSION TITLE
Tonal JS. How do you convert roman numeral notation to chord alias?

### DISCUSSION THREAD

**Discussion by Ant3ney** [tonaljs/tonal#296] [Q&A]
We are looking for a solution where we can pass in a roman numeral notation argument and get back a chord alias as a response. 

Basically, we are looking for a way to do the code below.
`let chordAlias = chordFromRoman("IVMaj9");`

We have tried the following.
```
let romemenNumeral = 'VIIMaj3';
let chordFromRomenNumerals = RomanNumeral.get(romemenNumeral).chordType
let chrodDetails = Chord.get(chordFromRomenNumerals);
console.log(chordDetails);
//logs
//{
//     aliases: []
//    chroma: ""
//    empty: true
//    intervals: []
//    name: ""
//    normalized: ""
//    notes: []
//    quality: "Unknown"
//    root: ""
//    rootDegree: 0
//    setNum: NaN
//    symbol: ""
//    tonic: null
//    type: ""
///}
```
The problem is that we keep getting back empty chords from `Chord.get(chordFromRomenNumerals)`

The only time `Chord.get(chordFromRomenNumerals)` returns something other than an empty set is when `let romemenNumeral = 'VIIMaj7';`. It seems as if the only supported chord types are Maj7 chords. Is that true?

Are we trying to do something that is not supported by Tonal JS? Is there some sort of package in Tonal JS that we are missing? 

Any help on this issue would be greatly supported.

**Comment by mrjacobbloom** [comment]
It seems like `chordType` returns the raw string after the roman numeral. So if there's an explicit chord type, you'll get that. But if there's not (it's a major or minor triad, implied from the capitalization), you'll get an empty string. BUT it'll always give you a boolean `major` property as well, so you can figure it out by combining them:

```
// With chordType:
'vii-7' -> { ... chordType: '-7' }
'IM9b5' -> { ... chordType: 'M9b5' }
'II+' -> { ... chordType: '+' }
'iii°' -> { ... chordType: '°' }

// Without chordType:
'ii' => { ... chordType: '', major: false }
'V' => { ... chordType: '', major: true }
```

So knowing that, you can fill in the missing part:

```js
const toChordType = (romanNumeral) => {
    const { chordType, major } = RomanNumeral.get(romanNumeral);
    if (chordType) return chordType;
    return major ? 'Major' : 'minor';
}
```

**Reply by mrjacobbloom** [reply]
@danigb this behavior isn't super intuitive IMO -- would you accept a PR to set `chordType` to "major" or "minor" when it'd be empty?

**Reply by Ant3ney** [reply]
> It seems like `chordType` returns the raw string after the roman numeral. So if there's an explicit chord type, you'll get that. But if there's not (it's a major or minor triad, implied from the capitalization), you'll get an empty string. BUT it'll always give you a boolean `major` property as well, so you can figure it out by combining them:
> 
> ```
> // With chordType:
> 'vii-7' -> { ... chordType: '-7' }
> 'IM9b5' -> { ... chordType: 'M9b5' }
> 'II+' -> { ... chordType: '+' }
> 'iii°' -> { ... chordType: '°' }
> 
> // Without chordType:
> 'ii' => { ... chordType: '', major: false }
> 'V' => { ... chordType: '', major: true }
> ```
> 
> So knowing that, you can fill in the missing part:
> 
> ```js
> const toChordType = (romanNumeral) => {
>     const { chordType, major } = RomanNumeral.get(romanNumeral);
>     if (chordType) return chordType;
>     return major ? 'Major' : 'minor';
> }
> ```

Thank you for the response.  I'll be implementing it soon.

**Reply by danigb** [reply]
@Ant3ney you can get the list of available chord names: `ChordDictionary.names()`

@mrjacobbloom  I see... yes, not very intuitive... but it has a reason: basically sometimes we need to know if the chord type is  specified or not. Also, when no chord type is specified, major or minor is assumed based on the letter case... but that's mostly a convention (as far as I know). So I'm not sure what is best
